### DISCUSSION TITLE
[Idea] Auto-Connect through opening a desired program

### DISCUSSION THREAD

**Discussion by Setnour6** [maximmax42/Discord-CustomRP#226] [Ideas]
This proposal allows for the Custom RP to automatically connect and disconnect when a program launches (example: Adobe After Effects, FL Studio), and load the presets for the program with the appropriate name.
The presets thing doesn't actually need to happen in the same procedure, but you can also work on that if you want to.

How it would work is that you can select a program or a list of programs to link with Custom RP so when they launch, Custom RP starts the connection with discord with the preset or the current settings to show on the rich presence. This is perfect for showing details on what you are doing, besides "Playing Adobe After Effects" and that's it.

Linking specific things (i.e. rendering example.mp4) won't need to be done. The user will still do that.

**Comment by maximmax42** [comment]
I mean it's a good idea, it's been suggested multiple times before, but I'm unsure how doable it is.

**Reply by Setnour6** [reply]
I haven't actually found suggestions like that here. Maybe it's the different keywords?
The image below is photoshopped but would have the idea. You can of course implement it in a different way (ex. a checkbox toggle):
![image](https://github.com/maximmax42/Discord-CustomRP/assets/70764728/7b42ea35-bb7d-40b6-b04a-c720a7a1deda)
The three dots would be where on clicking it, the file explorer opens to choose a program. The field would then display the path of the program or simply the program name with the path listed internally.

I am also including a draw.io file in a zipped folder that you can edit that includes a current and proposed depiction of the same program. It does look very robust since I had to make it quickly.
[Custom Rich Presence App Diagram.zip](https://github.com/maximmax42/Discord-CustomRP/files/13795214/Custom.Rich.Presence.App.Diagram.zip)

**Reply by maximmax42** [reply]
Well, that would only allow one presence to autoconnect at a time. To have multiple, I'd need to add some sort of list or something. And check the process list in the background. Not sure it's worth the hassle.

**Reply by Setnour6** [reply]
> Well, that would only allow one presence to autoconnect at a time. To have multiple, I'd need to add some sort of list or something. And check the process list in the background. Not sure it's worth the hassle.

We can start with one presence and see how it goes, then we can move on to a list of processes to link if we want. The path to the process will be part of the preset saving, so the path will always load and you won't have to re-select the process again unless it's not detected in the same location.

**Reply by MH285** [reply]
> Well, that would only allow one presence to autoconnect at a time. To have multiple, I'd need to add some sort of list or something. And check the process list in the background. Not sure it's worth the hassle.

would love to support you if you continue this!
